static/monorail-2.0-sb.yaml

```diff
@@ -158,29 +158,73 @@ paths:
             $ref: '#/definitions/Error'
   /notification/progress:
     x-swagger-router-controller: notification
+# In some environment (such as ipxe), the avaiable http client doesn't support POST, so we have
+# to compromise to define below GET api to update progress information, but actually it does the
+# job of verb "POST"
     get:
       operationId: notificationProgressGet
       summary: |
-        notification for task to get progress info
+        Send progress notification to task
       description: |
-        send taskid, total setps and current step sequence to rackhd via notification get function
+        This API is used to update the progress information for a task instance
       parameters:
         - name: taskId
           in: query
           description: |
-            Task instance identifier
-          required: true 
+            The identifier of task instance which the progress applies to
+          required: false
```

Comment:
Actually if the progress data is embedded in http request body, then the taskId is optional in query.